File: backend/src/main/java/pl/buczak/kacper/fleetmanagement/entity/dao/department/Department.java
package pl.buczak.kacper.fleetmanagement.entity.dao.department;

import pl.buczak.kacper.fleetmanagement.entity.dao.BaseEntity;
import pl.buczak.kacper.fleetmanagement.entity.dao.employee.Employee;

import javax.persistence.*;
import javax.validation.constraints.Size;
import java.util.List;

/*
    @author Kacper Buczak 
*/
@Entity
@Table(name = "department")
public class Department extends BaseEntity {

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id", referencedColumnName = "id")
    private Address address;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "region_id", referencedColumnName = "id")
    private Region region;

    @OneToMany(cascade = CascadeType.DETACH, mappedBy = "department")
    private List<Employee> employees;

    @Size(max = 50)
    @Column(nullable = false, name = "department_name")
    private String departmentName;

    public Department() {
    }

    public Address getAddress() {
        return address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }

    public Region getRegion() {
        return region;
    }

    public void setRegion(Region region) {
        this.region = region;
    }

    public List<Employee> getEmployees() {
        return employees;
    }

    public void setEmployees(List<Employee> employees) {
        this.employees = employees;
    }

    public String getDepartmentName() {
        return departmentName;
    }

    public void setDepartmentName(String departmentName) {
        this.departmentName = departmentName;
    }
}

File: frontend/src/app/shared/model/department/department.ts
import {Address} from '../address/address';
import {Region} from '../address/region';
import {Employee} from '../employee/employee';
import {BaseEntity, BaseEntityAttrs} from '../base-entity';

export interface DepartmentAttrs extends BaseEntityAttrs {
  address: Address;
  region: Region;
  employees: Employee[];
  departmentName: string;
}

export class Department extends BaseEntity {
  constructor(attrs: Partial<DepartmentAttrs> = {}) {
    super(attrs);
    this.address = attrs.address;
    this.region = attrs.region;
    this.employees = attrs.employees;
    this.departmentName = attrs.departmentName;
  }

  address: Address;
  region: Region;
  employees: Employee[];
  departmentName: string;
}
